android/src/main/java/com/polidea/flutter_ble_lib/FlutterBleLibPlugin.java

```diff
@@ -64,6 +76,12 @@ public void onMethodCall(MethodCall call, Result result) {
             case MethodName.STOP_DEVICE_SCAN:
                 stopDeviceScan(result);
                 break;
+            case MethodName.CONNECT_TO_DEVICE:
```

Comment:
what about asking a delegate if it is able to handle given method?

Reply:
Like a cascade of switches?
`if (xDelegate.canHandle(call))
    xDelegate.handleCall(call, result)
else if (yDelegate.canHandle(....
```

Reply:
`xDelegate.canHandle(call)` exactly.
But I am not sure `if.....else if` is the best solution. There are a lot of possibilities e.g.  to have a list of delegates and iterate over the list